src/OrchardCore/OrchardCore/Shell/ShellDescriptorFeaturesManager.cs

```diff
@@ -47,15 +50,15 @@ public class ShellDescriptorFeaturesManager : IShellDescriptorFeaturesManager
 
             var alwaysEnabledIds = _alwaysEnabledFeatures.Select(sf => sf.Id).ToArray();
 
-            var onDemandFeaturesToDisable = _extensionManager.GetFeatures()
+            var byDependencyOnlyFeaturesToDisable = enabledFeatures
                 .Where(f => f.EnabledByDependencyOnly);
 
             var allFeaturesToDisable = featuresToDisable
                 .Where(f => !alwaysEnabledIds.Contains(f.Id))
                 .SelectMany(feature => GetFeaturesToDisable(feature, enabledFeatureIds, force))
                 // Always attempt to disable EnabledByDependencyOnly features
                 // to ensure we auto disable any feature that is no longer needed
```

Comment:
For technical names we are used to add single quotes `'EnabledByDependencyOnly'`.

Add a dot at the end of the comment.

Hmm I would say `that has no more dependents`, but that's the same ;)